lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -434,10 +434,26 @@ def main(self):
         }
         if self.options.output:
             run_mpirun_cmd = RunFileLoopMPI.run
-            run_kwargs.update({'filename': self.options.output})
+            run_kwargs['filename'] = self.options.output
         else:
             run_mpirun_cmd = RunAsyncMPI.run
-        exitcode, _ = run_mpirun_cmd(self.mpirun_cmd, **run_kwargs)
+
+        if self.options.dry_run:
+            self.log.info("Dry run, only printing generated mpirun command...")
```

Comment:
move this to a function/method, so you can unittest it properly (instead of end2end)